internal/controllers/provisioningrequest_hwprovision.go

```diff
@@ -5,15 +5,64 @@ import (
 	"fmt"
 	"log/slog"
 
+	"k8s.io/apimachinery/pkg/api/equality"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	hwv1alpha1 "github.com/openshift-kni/oran-o2ims/api/hardwaremanagement/v1alpha1"
 	provisioningv1alpha1 "github.com/openshift-kni/oran-o2ims/api/provisioning/v1alpha1"
 	"github.com/openshift-kni/oran-o2ims/internal/controllers/utils"
 	siteconfig "github.com/stolostron/siteconfig/api/v1alpha1"
 )
 
+func (t *provisioningRequestReconcilerTask) createOrUpdateNodePool(ctx context.Context, nodePool *hwv1alpha1.NodePool) error {
+
+	existingNodePool := &hwv1alpha1.NodePool{}
+
+	exist, err := utils.DoesK8SResourceExist(ctx, t.client, nodePool.Name, nodePool.Namespace, existingNodePool)
+	if err != nil {
+		return fmt.Errorf("failed to get NodePool %s in namespace %s: %w", nodePool.GetName(), nodePool.GetNamespace(), err)
+	}
+
+	if !exist {
+		return t.createNodePoolResources(ctx, nodePool)
+	}
+
+	// The template validate is already completed; compare NodeGroup and update them if necessary
+	if !equality.Semantic.DeepEqual(existingNodePool.Spec.NodeGroup, nodePool.Spec.NodeGroup) {
+		// Only process the configuration changes if the cluster provisioning is fulfilled
+		if utils.IsProvisioningStateFulfilled(t.object) {
```

Comment:
These checks seem redundant since disallowed updates before fulfilled have been validated and rejected in the `CompareConfigMapWithNodePool` within `handleRenderHardwareTemplate`, so renderedHardwareTemplate condition will be failed and this function should not be reached.

Reply:
Removed.